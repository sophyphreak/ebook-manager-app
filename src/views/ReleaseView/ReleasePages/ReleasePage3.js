import React from 'react';
import { Card, CardHeader, CardBody, Button, Form } from 'reactstrap';

import Website from '../../FormElements/Website';
import Keywords from '../../FormElements/Keywords';
import ReleaseText from '../../FormElements/ReleaseText';

const ReleasePage3 = ({
  error,
  onBack,
  onSubmit,

  website,
  onWebsiteChange,

  keywords,
  onKeywordsChange,

  releaseText,
  onReleaseTextChange
}) => (
  <Card className="animated fadeIn">
    <CardHeader>
      <h4>
        <strong>Press Release - 3 of 3</strong>
      </h4>
    </CardHeader>
    <CardBody>
      <Form onSubmit={onSubmit}>
        <Website
          label="Website"
          website={website}
          onWebsiteChange={onWebsiteChange}
          hasError={!!error.message}
          errorMessage={error.website}
        />
        <Keywords
          label="Keywords"
          keywords={keywords}
          onKeywordsChange={onKeywordsChange}
          hasError={!!error.message}
          errorMessage={error.keywords}
        />
        <ReleaseText
          label="Press Release*"
          releaseText={releaseText}
          onReleaseTextChange={onReleaseTextChange}
          hasError={!!error.message}
          errorMessage={error.releaseText}
        />
        {error.message && <p>{error.message}</p>}
        <Button onClick={onBack} size="sm" color="basic">
          Go Back
        </Button>
        <Button type="submit" size="sm" color="primary" value="Submit">
          Submit
        </Button>
      </Form>
    </CardBody>
  </Card>
);

export default ReleasePage3;
