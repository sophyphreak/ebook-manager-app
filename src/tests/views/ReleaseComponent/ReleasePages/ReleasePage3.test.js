import React from 'react';
import { shallow } from 'enzyme';
import ReleasePage3 from '../../../../views/ReleaseComponent/ReleasePages/ReleasePage3';

let error, wrapper;

beforeEach(() => {
  error = {
    message: ''
  };
  wrapper = shallow(<ReleasePage3 error={error} />);
});

test('should render ReleasePage3 correctly', () => {
  expect(wrapper).toMatchSnapshot();
});
