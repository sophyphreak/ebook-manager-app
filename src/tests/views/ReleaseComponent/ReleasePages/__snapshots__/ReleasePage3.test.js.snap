// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ReleasePage3 correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ReleasePage3
    error={
      Object {
        "message": "",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <CardHeader
          tag="div"
        >
          <h4>
            <strong>
              Press Release - 3 of 3
            </strong>
          </h4>
        </CardHeader>,
        <CardBody
          tag="div"
        >
          <Form
            tag="form"
          >
            <Website
              hasError={false}
              label="Website"
            />
            <Keywords
              hasError={false}
              label="Keywords"
            />
            <ReleaseText
              hasError={false}
              label="Press Release*"
            />
            
            <Button
              color="basic"
              size="sm"
              tag="button"
            >
              Go Back
            </Button>
            <Button
              color="primary"
              size="sm"
              tag="button"
              type="submit"
              value="Submit"
            >
              Submit
            </Button>
          </Form>
        </CardBody>,
      ],
      "className": "animated fadeIn",
      "tag": "div",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <h4>
            <strong>
              Press Release - 3 of 3
            </strong>
          </h4>,
          "tag": "div",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <strong>
              Press Release - 3 of 3
            </strong>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Press Release - 3 of 3",
            },
            "ref": null,
            "rendered": "Press Release - 3 of 3",
            "type": "strong",
          },
          "type": "h4",
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <Form
            tag="form"
          >
            <Website
              hasError={false}
              label="Website"
            />
            <Keywords
              hasError={false}
              label="Keywords"
            />
            <ReleaseText
              hasError={false}
              label="Press Release*"
            />
            
            <Button
              color="basic"
              size="sm"
              tag="button"
            >
              Go Back
            </Button>
            <Button
              color="primary"
              size="sm"
              tag="button"
              type="submit"
              value="Submit"
            >
              Submit
            </Button>
          </Form>,
          "tag": "div",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <Website
                hasError={false}
                label="Website"
              />,
              <Keywords
                hasError={false}
                label="Keywords"
              />,
              <ReleaseText
                hasError={false}
                label="Press Release*"
              />,
              "",
              <Button
                color="basic"
                size="sm"
                tag="button"
              >
                Go Back
              </Button>,
              <Button
                color="primary"
                size="sm"
                tag="button"
                type="submit"
                value="Submit"
              >
                Submit
              </Button>,
            ],
            "onSubmit": undefined,
            "tag": "form",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "errorMessage": undefined,
                "hasError": false,
                "label": "Website",
                "onWebsiteChange": undefined,
                "website": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "errorMessage": undefined,
                "hasError": false,
                "keywords": undefined,
                "label": "Keywords",
                "onKeywordsChange": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "errorMessage": undefined,
                "hasError": false,
                "label": "Press Release*",
                "onReleaseTextChange": undefined,
                "releaseText": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Go Back",
                "color": "basic",
                "onClick": undefined,
                "size": "sm",
                "tag": "button",
              },
              "ref": null,
              "rendered": "Go Back",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Submit",
                "color": "primary",
                "size": "sm",
                "tag": "button",
                "type": "submit",
                "value": "Submit",
              },
              "ref": null,
              "rendered": "Submit",
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <CardHeader
            tag="div"
          >
            <h4>
              <strong>
                Press Release - 3 of 3
              </strong>
            </h4>
          </CardHeader>,
          <CardBody
            tag="div"
          >
            <Form
              tag="form"
            >
              <Website
                hasError={false}
                label="Website"
              />
              <Keywords
                hasError={false}
                label="Keywords"
              />
              <ReleaseText
                hasError={false}
                label="Press Release*"
              />
              
              <Button
                color="basic"
                size="sm"
                tag="button"
              >
                Go Back
              </Button>
              <Button
                color="primary"
                size="sm"
                tag="button"
                type="submit"
                value="Submit"
              >
                Submit
              </Button>
            </Form>
          </CardBody>,
        ],
        "className": "animated fadeIn",
        "tag": "div",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <h4>
              <strong>
                Press Release - 3 of 3
              </strong>
            </h4>,
            "tag": "div",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <strong>
                Press Release - 3 of 3
              </strong>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Press Release - 3 of 3",
              },
              "ref": null,
              "rendered": "Press Release - 3 of 3",
              "type": "strong",
            },
            "type": "h4",
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <Form
              tag="form"
            >
              <Website
                hasError={false}
                label="Website"
              />
              <Keywords
                hasError={false}
                label="Keywords"
              />
              <ReleaseText
                hasError={false}
                label="Press Release*"
              />
              
              <Button
                color="basic"
                size="sm"
                tag="button"
              >
                Go Back
              </Button>
              <Button
                color="primary"
                size="sm"
                tag="button"
                type="submit"
                value="Submit"
              >
                Submit
              </Button>
            </Form>,
            "tag": "div",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <Website
                  hasError={false}
                  label="Website"
                />,
                <Keywords
                  hasError={false}
                  label="Keywords"
                />,
                <ReleaseText
                  hasError={false}
                  label="Press Release*"
                />,
                "",
                <Button
                  color="basic"
                  size="sm"
                  tag="button"
                >
                  Go Back
                </Button>,
                <Button
                  color="primary"
                  size="sm"
                  tag="button"
                  type="submit"
                  value="Submit"
                >
                  Submit
                </Button>,
              ],
              "onSubmit": undefined,
              "tag": "form",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "errorMessage": undefined,
                  "hasError": false,
                  "label": "Website",
                  "onWebsiteChange": undefined,
                  "website": undefined,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "errorMessage": undefined,
                  "hasError": false,
                  "keywords": undefined,
                  "label": "Keywords",
                  "onKeywordsChange": undefined,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "errorMessage": undefined,
                  "hasError": false,
                  "label": "Press Release*",
                  "onReleaseTextChange": undefined,
                  "releaseText": undefined,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "Go Back",
                  "color": "basic",
                  "onClick": undefined,
                  "size": "sm",
                  "tag": "button",
                },
                "ref": null,
                "rendered": "Go Back",
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "Submit",
                  "color": "primary",
                  "size": "sm",
                  "tag": "button",
                  "type": "submit",
                  "value": "Submit",
                },
                "ref": null,
                "rendered": "Submit",
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
